### PR TITLE
drivers: ieee802154: Develop 15.4 native_posix radio_api with BabbleSim

### PR DESCRIPTION
**Pull request by FilipZajdel**
Adds native_posix radio hardware model built using BabbleSim
as well as ieee802.15.4 radio api implementation. This component
enables performing tests of higher layers of networking stacks.
Due to lack 802.15.4 physical layer support in BabbleSim,
some features (like lqi, rssi) are missing. The communication
between devices is made with phy_2G4_libPhyComv1 using workarounds
(all devices share the same phy address) so phy2G4 can see packets
as they were Ble packets. Frames validation, filtering, ack generation
etc. in ieee802.15.4 radio_api driver is made using the code from
ieee802.15.4_nrf5. Sample samples/ieee802154 was made for testing
the native_posix 15.4 driver and consists of two programs: device0
and device1, which exchange 802.15.4 frames.

Signed-off-by: Filip Zajdel <filip.zajdel@nordicsemi.no>

### CODE REVIEW

**Comment by FilipZajdel** [comment]
Files with prefix nrf_802154_* were copied from https://github.com/nrfconnect/sdk-hal_nordic.

**Review by aescolar — CHANGES_REQUESTED** [review]
2 comments just from a very quick scroll. (This will require a long review)

File: boards/posix/native_posix/cmdline.c [hunk 1]
```diff
@@ -118,16 +118,19 @@ void native_handle_cmd_line(int argc, char *argv[])
 	native_add_tracing_options();
 	native_add_testargs_option();
 
+#if CONFIG_USE_BABBLESIM
+	bs_radio_argparse_add_options();
```
[the remark this answers is not in the document]
Comment (FilipZajdel) [reply boards/posix/native_posix/cmdline.c]: Done! I used NATIVE_TASK macro for this purpose.

File: boards/posix/native_posix/cmdline.c [hunk 2]
```diff
@@ -54,9 +55,9 @@ void native_add_command_line_opts(struct args_struct_t *args)
 			growby = ARGS_ALLOC_CHUNK_SIZE;
 		}
 
-		struct args_struct_t *new_args_struct = realloc(args_struct,
-				      (args_aval + growby)*
-				      sizeof(struct args_struct_t));
+		struct args_struct_t *new_args_struct = realloc(
```
[the remark this answers is not in the document]
Comment (FilipZajdel) [reply boards/posix/native_posix/cmdline.c]: Done! I removed all changes I made to this file.

**Comment by FilipZajdel** [comment]
@aescolar I pushed some changes regarding your comments.

**Review by tbursztyka — CHANGES_REQUESTED** [review]
You will need to split this into various commits.

And also: explain better what you are trying to achieve.

What is BabbleSim? Why is nrf_ prefix used almost everywhere in files that are supposed to be native_posix? etc...

**Comment by FilipZajdel** [comment]
@tbursztyka 
Thank you, next PRs will have better explanation. 

The thing I am working on is developing the generic emulation of radio which lives in native_posix board and uses BabbleSim (physical layer simulator). The solution similar to nrf52_bsim board, but more generic; without emulation of registers etc. Then I'm going to use it for ieee802154 radio_api implementation, and finally it can be used to simulate Zigbee (and Thread maybe) using PC with no need of development kits.

I will split it into 3 commits:
1. will add bs_radio module in native posix board.
2. will add ieee802154 driver.
3. will add sample application.

Does it sound relevant?

**Pull request closed without merging**